File: format.py
from json import dump
from os import listdir
from os.path import join
from re import match, sub
from xml.etree.ElementTree import Element, tostring
from string import ascii_letters
from quiz import generate

for file in listdir('input'):
    with open(join('input', file), 'r', encoding='utf-8') as f:
        content = f.read()
    content = content.split('\n')
    current = None
    result = []

    for line in content:
        question = match(r'^\s*[^\d]*(\d+)\.(.*)$', line)
        if question and not current:
            current = [[question.group(1), question.group(2).strip()]]
            continue

        variant = match(r'^([A-EА-Я])\.(.*)$', line)
        if variant:
            variant = [variant.group(1), variant.group(2).strip()]
            current.append(variant)

        variant = match(r'^\s+ANSWER.*(A-E)', line)
        if variant:
            current.append(variant.group(1))
            result.append(current)
            continue

    generate(result, file)

File: format.js
var fs = require('fs');
var exec = require('child_process').exec;

var dir = fs.readdirSync('input');
for(var i in dir) {
    var file = fs.readFileSync('input/' + dir[i]);
    file = file.toString();
    var result = [];
    var content = file.split('\n');
    for(var j=0; j<content.length; j++) {
        var line = content[j];
//        line = line.replace('’', "'");
        line = line.replace(/\s+[\.,]/, function(s) {
            return s[s.length - 1];
        });

        var q = /^\s*(\d+)\.(.*)/.exec(line);
        if (q) {
            var current = [];
            current.push(q.slice(1));
        }
        var v = /^([A-EА-Я])\.(.*)/.exec(line);
        if (v)
            current.push(v.slice(1));
//        fs.writeFileSync('count', k);
        var a = /ANSWER:\s*([A-E])/.exec(line);

        if (a) {
            current.push(a.slice(1)[0]);
            result.push(current);
        }

    }
    file = JSON.stringify(result, null, '\t');
    var json = 'json/' + dir[i];
    fs.writeFileSync(json + '.json', file);
    fs.appendFileSync('count', json + ' ' + result.length + '\n');
    exec('python quiz.py ' + json + ' html/' + dir[i]);
}
